src/cocotb/runner.py

```diff
@@ -363,13 +369,27 @@ def _get_parameter_options(self, parameters: Mapping[str, object]) -> Command:
     def _execute(self, cmds: Sequence[Command], cwd: PathLike) -> None:
         __tracebackhide__ = True  # Hide the traceback when using PyTest.
 
+        if self.log_file is None:
+            self._execute_cmds(cmds, cwd)
+        else:
+            with open(self.log_file, "w") as f:
+                self._execute_cmds(cmds, cwd, f)
+
+    def _execute_cmds(
+        self, cmds: Sequence[Command], cwd: PathLike, stdout: Optional[TextIO] = None
+    ) -> None:
+        __tracebackhide__ = True  # Hide the traceback when using PyTest.
+
         for cmd in cmds:
             print(f"INFO: Running command {shlex_join(cmd)} in directory {cwd}")
 
             # TODO: create a thread to handle stderr and log as error?
             # TODO: log forwarding
```

Comment:
Leaving this because I assume "log forwarding" means capturing stdout / stderr and yielding that as part of the `build()` or `test()` call, which is not what this PR is doing.